cylc/flow/suite_files.py

```diff
@@ -0,0 +1,723 @@
+
+    These describe properties of a running suite.
+
+    .. note::
+
+       The presence of this file indicates that the suite is running as it is
+       removed when a suite shuts-down, however, in exceptional circumstances,
+       if a suite is not properly shut-down this file may be left behind.
+
+    """
+
+    API = 'CYLC_API'
+    """The Suite API version string."""
+
+    COMMS_PROTOCOL_2 = 'CYLC_COMMS_PROTOCOL_2'  # indirect comms
+
+    HOST = 'CYLC_SUITE_HOST'
+    """The name of the host the suite server process is running on."""
+
+    NAME = 'CYLC_SUITE_NAME'
+    """The name of the suite."""
+
+    OWNER = 'CYLC_SUITE_OWNER'
+    """The user account under which the suite server process is running."""
+
+    PROCESS = 'CYLC_SUITE_PROCESS'
+    """The process ID of the running suite on ``CYLC_SUITE_HOST``."""
+
+    PORT = 'CYLC_SUITE_PORT'
+    """The port Cylc uses to communicate with this suite."""
+
+    SSH_USE_LOGIN_SHELL = 'CYLC_SSH_USE_LOGIN_SHELL'
+    """TODO: Unused at present, waiting on #2975."""
+
+    SUITE_RUN_DIR_ON_SUITE_HOST = 'CYLC_SUITE_RUN_DIR_ON_SUITE_HOST'
+    """The path to the suite run directory as seen from ``HOST``."""
+
+    TASK_MSG_MAX_TRIES = 'CYLC_TASK_MSG_MAX_TRIES'
+    """TODO: Unused at present, waiting on #3331."""
+
+    TASK_MSG_RETRY_INTVL = 'CYLC_TASK_MSG_RETRY_INTVL'
+    """TODO: Unused at present, waiting on #3331."""
+
+    TASK_MSG_TIMEOUT = 'CYLC_TASK_MSG_TIMEOUT'
+    """TODO: Unused at present, waiting on #3331."""
+
+    UUID = 'CYLC_SUITE_UUID'
+    """Unique ID for this run of the suite."""
+
+    VERSION = 'CYLC_VERSION'
+    """The Cylc version under which the suite is running."""
+
+
+class UserFiles:
+    """Directory containing config and auth files for a user."""
+
+    DIRNAME = '.cylc'
+    """The name of this directory."""
+
+    class Auth:
+        """Cache for remote service files."""
+
+        DIRNAME = 'auth'
+        """The name of this directory."""
+
+    @classmethod
+    def get_path(cls):
+        """Return the path to this directory for the current user."""
+        return os.path.join(
+            os.path.expanduser("~"),
+            cls.DIRNAME,
+            cls.Auth.DIRNAME,
+        )
+
+
+REG_DELIM = "/"
+
+NO_TITLE = "No title provided"
+REC_TITLE = re.compile(r"^\s*title\s*=\s*(.*)\s*$")
+
+PASSPHRASE_CHARSET = ascii_letters + digits
+PASSPHRASE_LEN = 20
+
+PS_OPTS = '-opid,args'
+
+CONTACT_FILE_EXISTS_MSG = r"""suite contact file exists: %(fname)s
+
+Suite "%(suite)s" is already running, and listening at "%(host)s:%(port)s".
+
+To start a new run, stop the old one first with one or more of these:
+* cylc stop %(suite)s              # wait for active tasks/event handlers
+* cylc stop --kill %(suite)s       # kill active tasks and wait
+
+* cylc stop --now %(suite)s        # don't wait for active tasks
+* cylc stop --now --now %(suite)s  # don't wait
+* ssh -n "%(host)s" kill %(pid)s   # final brute force!
```

Comment:
I noticed the deprecation/pending removal of `SSH_USE_LOGIN_SHELL`, and #2975, does that mean here this example should be just `kill %(pid)s`?

Reply:
Will still need to SSH to the host to kill the suite server process.